aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -645,7 +613,7 @@ SparseTensor& hspmm_out_sparse_cpu(SparseTensor& r, const SparseTensor& sparse_,
   AT_CHECK(sparse_._denseDims() == 0,
       "Argument #2: scalar values expected, got ", sparse_._denseDims(), "D values");
   AT_CHECK(dense.dim() == 2,
-      "Argument #2: matrices expected, got ", dense.dim(), "D tensor");
+      "Argument #3: matrices expected, got ", dense.dim(), "D tensor");
```

Comment:
these are probably not safe with n-dimensional empty tensors -- make this `!dense.is_empty() && dense.dim() == 2` (or I can later).

Reply:
Alas, you haven't added `is_empty` as a method on `at::Tensor` yet :)

Reply:
it's there now :).